packages/advanced-logic/specs/payment-network-erc20-fee-proxy-contract-0.1.0.md

```diff
@@ -0,0 +1,328 @@
+
+The 'addRefundAddress' event:
+
+|  Property                    |  Value                          |
+| ---------------------------- | ------------------------------- |
+| **name**                     | 'addRefundAddress'              |
+| **parameters**               |                                 |
+| **parameters.refundAddress** | `refundAddress` from parameters |
+
+#### addFeeAddress
+
+##### Parameters
+
+|                           | Type   | Description                                   | Requirement   |
+| ------------------------- | ------ | --------------------------------------------- | ------------- |
+| **id**                    | String | Constant value: "pn-erc20-fee-proxy-contract" | **Mandatory** |
+| **action**                | String | Constant value: "addFeeAddress"               | **Mandatory** |
+| **parameters**            | Object |                                               |               |
+| **parameters.feeAddress** | String | Ethereum address for the fee payment          | **Mandatory** |
+
+##### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc20-fee-proxy-contract" exists
+- The signer is the `payee`
+- The extension property `feeAddress` is undefined
+
+##### Warnings
+
+None.
+
+##### Results
+
+An extension state is updated with the following properties:
+
+|  Property             |  Value                                          |
+| --------------------- | ----------------------------------------------- |
+| **values.feeAddress** | `feeAddress` from parameters                    |
+| **events**            | Add a 'feeAddress' event (see below) at its end |
+
+the 'addFeeAddress' event:
+
+|  Property                 |  Value                          |
+| ------------------------- | ------------------------------- |
+| **name**                  | Constant value: "addFeeAddress" |
+| **parameters**            |                                 |
+| **parameters.feeAddress** | `feeAddress` from parameters    |
+
+#### addFeeAmount
+
+##### Parameters
+
+|                          | Type   | Description                                   | Requirement   |
+| ------------------------ | ------ | --------------------------------------------- | ------------- |
+| **id**                   | String | Constant value: "pn-erc20-fee-proxy-contract" | **Mandatory** |
+| **action**               | String | Constant value: "addFeeAmount"                | **Mandatory** |
+| **parameters**           | Object |                                               |               |
+| **parameters.feeAmount** | String | The fee amount                                | **Mandatory** |
+
+##### Conditions
+
+This action is valid, if:
+
+- The extension state with the id "pn-erc20-fee-proxy-contract" exists
+- The signer is the `payee`
+- The extension property `feeAmount` is undefined
+
+##### Warnings
+
+This action must trigger the warnings:
```

Comment:
Missing a 2nd warning if the method is called by the payer

Reply:
Same as previous.